src/main/java/racingCar/RacingCar.java

```diff
@@ -0,0 +1,175 @@
+		try {
+			checkEndBlank(input);
+		} catch (RuntimeException e) {
+			return racingCarNames();
+		}
+		return makeCarNames(input);
+	}
+
+	private void checkEndBlank(String input) throws RuntimeException{
+		if (input.endsWith(",")) {
+			System.out.println(NAME_ERROR);
+			throw new RuntimeException();
+		}
+	}
+
+	private void racingCarTimes() {
+		String inputRacingTimes = userInput(WHAT_TIMES);
+		enterTimes(inputRacingTimes);
+	}
+
+	private String userInput(String message) {
+		System.out.println(message);
+		Scanner scanner = new Scanner(System.in);
+		return scanner.nextLine();
+	}
+
+	public String[] makeCarNames(String input) {
+		String[] carNames = splitCarNames(input);
+		return getCarNames(carNames);
+	}
+
+	private String[] getCarNames(String[] carNames) {
+		for (int idx = 0; idx < carNames.length; idx++) {
+			carNames[idx] = carNames[idx].trim();
+			checkCarName(carNames[idx]);
+		}
+		return carNames;
+	}
+
+	private String[] splitCarNames(String input) {
+		String[] carNames = input.split(COMMA);
+		return carNames;
+	}
+
+	private void checkCarName(String name) {
+		try {
+			checkRightLength(name);
+		} catch (RuntimeException e) {
+			System.out.println(NAME_ERROR);
+			racingCarNames();
+		}
+	}
+
+	public void checkRightLength(String name) throws RuntimeException {
+		if (isRightLength(name)) {
+			throw new RuntimeException();
+		}
+	}
+
+	private boolean isRightLength(String name) {
+		return name.length() == 0 || name.length() > NAME_MAX_LENGTH;
+	}
+
+	public void enterTimes(String input) {
+		try {
+			isRightTimes(input);
+			times = Integer.parseInt(input);
+		} catch (RuntimeException e) {
+			System.out.println(TIMES_ERROR);
+			racingCarTimes();
+		}
+	}
+
+	public void isRightTimes(String input) throws RuntimeException {
+		final String REGEX = "[0-9]+";
+		if (!input.matches(REGEX)) {
+			throw new RuntimeException();
+		}
+	}
+
+	private void makeCars(String[] carNames) {
+		for (String carName : carNames) {
+			cars.add(new Car(carName));
+		}
+	}
+
+	private void race(int count) {
+		System.out.println(RESULT);
+		for (int i = 0; i < count; i++) {
+			moveCars();
+			System.out.println();
+		}
+	}
+
+	private void moveCars() {
+		for (int idx = 0; idx < cars.size(); idx++) {
+			moveCar(idx);
+		}
+	}
+
+	private void moveCar(int idx) {
+		Car car = cars.get(idx);
+		car.moveCar(makeRandom());
+		System.out.println(car);
+	}
+
+	private boolean makeRandom() {
+		return ((int)(Math.random() * 10) - 1) >= ZERO_TO_NINE_PROBABILITY;
+	}
+
+	public List<Car> findWinner(List<Car> cars) {
+		List<Car> winners = new ArrayList<>();
+		for (Car car : cars) {
+			winners = checkLead(car, winners);
+		}
+		return winners;
+
+	}
+
+	private List<Car> checkLead(Car car, List<Car> winners) {
+		if (car.getPosition() > maxPosition) {
+			winners.clear();
+			winners.add(car);
+			maxPosition = car.getPosition();
+			return winners;
+		}
+		if (car.getPosition() == maxPosition) {
```

Comment:
객체에게 메시지를 보내라!
기억하시나요? 프리코스에서 피드백으로 나왔을거에요. 기억나지 않으신다면 [링크](https://tecoble.techcourse.co.kr/post/2020-04-28-ask-instead-of-getter/)를 참고해주세요.

`Car` 객체에게 너가 특정한 포지션과 같은 포지션을 가지는 자동차인지 메시지를 보내보는건 어떨까요?

Reply:
같을때만 바꿔볼려고 Car 클래스에 메서드를 추가했어요.! 이런식으로 더 클 때 getter 없이 max값도 찾는 건가요?
```suggestion
		public boolean checkSamePosition(int maxPosition) {
		return this.position == maxPosition;
	}
```

Reply:
지금 레오가 리팩토링한 부분은 이번 미션의 핵심이라 생각해요...!!

프리코스 때에 배웠던, 객체에게 메시지를 보내는 방식으로 수정한거에요. 

Car 객체 내부의 position을 게터로 가져와서 이를 RacingCar에서 maxPosition과 비교하지 않고말이죠!

이제는 Car 객체가 스스로 나의 위치가 입력받은 position과 같은지를 확인해서 메시지에 응답하는 능동적인 행위를 합니다!